rspace_client/notebook_sync/sync_notebook.py

```
@@ -0,0 +1,498 @@
+            return {'name': nb_fname + ext, 'root_name': nb_fname, 'name_path': nb_path}
+        except Exception as e:
+            print(f"Error getting notebook name, please manually set a value for 'notebook_name'")
+            raise
+
+    def get_password():
+        """
+        Retrieves password from (or saves a new password to) keyring
+        """
+        service_id = RSPACE_KEYRING_SERVICE_ID
+
+        retrieved_password = keyring.get_password(service_id, rspace_username)
+        if retrieved_password is None:
+            retrieved_password = getpass.getpass("Please enter your RSpace Api key: ")
+            keyring.set_password(service_id, username, retrieved_password)
+        return retrieved_password
+
+    def get_rspace_client():
+        """
+        Returns rspace ELN API client
+        """
+        global rspace_client
+        if rspace_client is None:
+            retrieved_password = get_password()
+            rspace_client = eln.ELNClient(rspace_url, retrieved_password)
+        return rspace_client
+
+    def save_rspace_data(rspace_doc, attachments, gallery_file, execution_count, history_data):
+        # Define the filename to save the state
+        state_filename = get_notebook_name()['root_name'] + "_state.pkl"
+        with open(state_filename, 'wb') as f:
+            dill.dump({RSPACE_DOC_FOR_NOTEBOOK: rspace_doc, RSPACE_ATTACHMENTS_FOR_NOTEBOOK: attachments,
+                       RSPACE_GALLERY_FILE_FOR_NOTEBOOK: gallery_file,
+                       RSPACE_EXECUTION_COUNT_FOR_NOTEBOOK: execution_count, RSPACE_HISTORY_DATA: history_data}, f)
+
+    def load_data():
+        state_filename = get_notebook_name()['root_name'] + "_state.pkl"
+
+        if os.path.exists(state_filename):
+            # Load the variables from the file using dill
+            with open(state_filename, 'rb') as f:
+                try:
+                    loaded_state = dill.load(f)
+                except Exception as e:
```

Should the sync really fail silently?